esp-mbedtls/src/esp_hal.rs

```diff
@@ -0,0 +1,68 @@
+#[doc(hidden)]
+use core::cell::RefCell;
+use critical_section::Mutex;
+
+use esp_hal::peripheral::Peripheral;
+use esp_hal::peripherals::{RSA, SHA};
```

Comment:
Not sure. To "not depend" on the `esp-hal` crate in practice don't you actually need to re-export more from `esp-hal`, not just `Peripheral`, `RSA` and `SHA`? For example, I'm not sure if `esp-hal` uses the `RSA` and `SHA` types directly, or has some traits on these that need to be re-exported too?

... and then in practice if you plan to use `esp-hal` with `esp-mbedtls` you anyway will use a wider surface of `esp-hal` in your real-life app as you probably need quite a lot from it for the other aspects of your embedded project. And moreover, the `esp-hal` version used by `esp-mbedtls` probably needs to be **the exact same version** as the one that the user would directly depend on in their `Cargo.toml` (due to peripheral singletons etc.)

In such circumstances I usually do one of two things:
- Option A: Re-export ALL of the dependent crate under some sort of module. As in `pub mod esp_mbedtls::esp_hal { use esp_hal::*; }`. This is what we do with `esp-idf-svc` -> `esp-idf-hal` -> `esp-idf-sys` where each "more top-level" crate re-exports the lower-level crate under a module, e.g. `esp_idf_svc::hal`. Yes, I did this in `esp-mbedtls` in that it re-exports the `embedded-io` crates, but these have a much smaller surface than `esp-hal` and for those it might happen (a) that the user depends on multiple versions of those which is OK (b) the user does **not** have explicit dependency on embedded-io elsewhere in their project (though unlikely, but then again, multiple versions of embedded-io is OK, unlike esp-hal). (BTW, whatever we do, we probsably need to re-export `esp-mbedtls-sys` as `::sys`)
- Option B: Do not re-export anything

I think for `esp-mbedtls` given that it is having a more niche use-case than `esp-hal` it should rather not re-export anything, i.e. Option B?

Reply:
BTW, if you plan to move away from `esp-hal` (and this PR is in fact doing this already, as it supports other environments) shouldn't you rename it actually?

`edge-mbedtls` might be a good name, and if you wish, we can merge it inside `edge-net` and you'll get commiter privs there :p 

Or you just pick up another name and keep it separate.

Reply:
> BTW, if you plan to move away from `esp-hal` (and this PR is in fact doing this already, as it supports other environments) shouldn't you rename it actually?
> 
> `edge-mbedtls` might be a good name, and if you wish, we can merge it inside `edge-net` and you'll get commiter privs there :p
> 
> Or you just pick up another name and keep it separate.

Haha, I'm just a contributor here. It's up to the maintainers at @esp-rs to decide what is the future of this crate here, and where it should go. Currently, I see it as a bandaid patch, until we get a pure-rust solution working, like `rustls`.